File: bad_code_from_long_ago/samara2012/d.cpp
#include <bits/stdc++.h>
using namespace std;
#define _USE_MATH_DEFINES

//#define DEBUG
//#define USE_MAGIC_IO

#define ll long long
#define pii pair<int,int>
#define pdd pair<double,double>
#define ldouble long double
#define nl '\n'
const ll MOD = 1e9+7;
const int INF = 0x3f3f3f3f;
const ll INFLL = 0x3f3f3f3f3f3f3f3f;

ll dp[1000001];
ll solve(int n, const vector<int>& factors) {
	if (n == 1) return 1;
	if (dp[n] != -1)
		return dp[n];
	dp[n] = 0;
	for (int f: factors) {
		if (f > n) break;
		if (n % f == 0)
			dp[n] += solve(n/f, factors);
	}
	return dp[n];
}

int main() {
	ios::sync_with_stdio(0); 
	cin.tie(0); cout.tie(0);

	vector<int> factors;
	int n;
	cin >> n;
	for (int i = 2; i <= n; i++)
		if (n % i == 0)
			factors.push_back(i);
	memset(dp, -1, sizeof(dp));
	cout << solve(n, factors) << nl;
 
	return 0;
}

File: bad_code_from_long_ago/samara2012/d.py

vector<int> primes;
def getprimes(n):
  primes = []
  bitset<10000000> notprime;
  int lim = sqrt(n);
  notprime[0] = notprime[1] = true;
  for (int i = 2; i <= lim; i++)
    if (notprime[i])
      for (int j = i*i; j <= n; j++)
        notprime[j] = true;
  for (int i = 0; i <= n; i++)
    if (!notprime[i])
      primes.push_back(i);
}

vector<int> primefactor(int n) {
  vector<int> res;
  int count;
  int lim = sqrt(n);
  for (int i = 0; primes[i] <= lim; i++) {
    if (n % primes[i] == 0) {
      count = 0;
      while (n % primes[i] == 0) {
        n /= primes[i];
        count++;
      }
      res.push_back(count);
      lim = sqrt(n);
    }
  }
  if (n > 1) {
    res.push_back(1);
  }
  return res;
}

int main() {
  ios::sync_with_stdio(0); 
  cin.tie(0); cout.tie(0);

  vector<int> pf;
  int n;
  cin >> n;
  pf = primefactor(n);
  int sum = 0;
  for (int it: pf) 
    sum += it;
  ll ans = 1;
  for (int i = 1; i <= sum; i++) {
    ans *= i;
  }

  return 0;
}
